### DISCUSSION TITLE
Payment response with PaymentSheet

### DISCUSSION THREAD

**Discussion by DevFreeze** [flutter-stripe/flutter_stripe#223] [Q&A]
How do I get a response of the payment with the PaymentSheet, if it was successful or not?

The function 'presentPaymentSheet' only returns a void. This means, even after closing the sheet it returns a successful payment (according to the example app).

**Comment by jonasbark** [comment]
> This means, even after closing

What do you mean by closing? If you close the sheet without making a payment then you'll get the information that the payment was canceled. If the method succeeds then your payment succeeded.

**Reply by pateldhiraj782** [reply]
from where we will get that payment is succeeded or cancelled ?

**Reply by remonh87** [reply]
In the upcoming version we will throw an `StripeException`  (in accordance with the behavior on the stripe sdk)  that contains a code. Consumers should catch this exception. See https://github.com/flutter-stripe/flutter_stripe/blob/e50d4a5158651a2ef8a87b29b318980d926d790b/packages/stripe/lib/src/stripe.dart#L305
